### ISSUE TITLE
Smart replace with "null" on php8 constructor promoted property

### ISSUE THREAD

**Issue by asgraf** [hammer-tools/php-hammer#32]
Example:
```php
class SomeClass
{
    public function __construct(private $variable = 'default') {
    }
}
```
Aplying `Smart replace with "null"` generates:
```php
class SomeClass
{
    public function __construct(private $variable = null) {
        $variable ??= 'default',
    }
}
```
In case of promoted properties generated code should look like this:
```php
class SomeClass
{
    public function __construct(private $variable = null) {
        $this->variable ??= 'default',
    }
}
```

More info:
https://php.watch/versions/8.0/constructor-property-promotion

**Comment by asgraf** [comment]
Related issue #33

**Comment by rentalhost** [comment]
Thank you for your time in reporting this issue. The fix has been implemented in **version 0.25.0** and is **available exclusively** through the [**Releases**](https://github.com/hammer-tools/php-hammer/releases) page. It will be implemented in the JetBrains Marketplace once [**this issue**](https://github.com/hammer-tools/php-hammer/issues/10) that are affecting some users are resolved (no ETA).

**In response to your report:**

- The issue has been identified and fixed. Now, when applying the suggested fix, `$this` will be included for promoted properties. (eg. `$this->variable ??= 'default'`)
